EIPS/EnsIP01.md

```diff
@@ -0,0 +1,61 @@
+
+7) **Invalid bids** one extra factor would go into checking if a bid was valid: if the bid was revealed and the private key was larger than the size they had revealed, then the bid is declared invalid. Users are free to pick keys larger than the difficulty if they want to reveal it themselves, but they must be honest on the minimum size, so that revealers can choose to ignore bids that would take longer than 24 hours.
+
+7) **Rewards origin** the system can pay itself, not only by requiring deposit fees but also by removing the **burning** of fees that occur in the current system. If a bid is revealed in 48h or less, then the reveal fee is paid directly from the bidder's deposit. The larger reveal fees, paid when a bid isn't revealed after 48h are taken from the larger fee pile, because it's not necessarily the bidder's fault. Bids encrypted with a larger key than the suggested difficulty pay no revealer's fee.
+
+## Possible issues
+
+* The process of bruteforcing keys could be considered a waste of electricity and a step back to the environment just as ethereum moves to a less resource intensive consensus engine.
+* There is a non-trivial coordination issue when miners are deciding which bids to work on. If it takes 48h of constant bruteforcing to find a key, then all is wasted if another miners finds the same key before you. Miners could coordinate to attempt to bruteforce different bids, but this could create an unfair competition among parties that trust themselves not to lie.
+* If there is enough collusion and not enough competition for bids, revealers can inflate the reward by finding bids and not revealing them until later when the rewards are larger.
+* If there's not enough competition, revealers can find a way to bids for fake names just for the purpose of draining the ether on the fee accounts
+
+
+## Conclusion
+
+In the proposed system, the users would only need to do one transaction. Since the message is signed locally on a private key created on the client, that part can be also invisible to the user. All bids are revealed automatically for those who opted in the system, but can be also revealed manually if the user so desires (they can decide to give the key to a trusted third party to reveal it). This also would give a more useful utility for the thousands of ethers that have been currently [burned in the bidding fees](https://etherscan.io/address/0x000000000000000000000000000000000000dead), and could create another source of revenue for old mining machines.
+
+
+I welcome discussions and ideas on this area: there might be some other more clever uses of zk-snarks that can accomplish the same result without having to bruteforce keys.
+
+## Acknowledgements
+
+I'd like to thank Nick Johnson for organizing the ENS workshop in London and all those who have attended for the very useful debates
+
```

Comment:
Please waive the copyright, like in https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1.md#copyright